driver/ppm_fillers.c

```diff
@@ -5094,6 +5098,67 @@ static int f_sys_semop_x(struct event_filler_arguments *args)
 	return add_sentinel(args);
 }
 
+static inline u32 semget_flags_to_scap(unsigned flags)
+{
+	u32 res = 0;
+
+	if (flags & IPC_CREAT)
+		res |= PPM_IPC_CREAT;
+
+	if (flags & IPC_EXCL)
+		res |= PPM_IPC_EXCL;
+
+	return res;
+}
+
+static int f_sys_semget_e(struct event_filler_arguments *args)
+{
+	unsigned long val;
+	int res;
+
+	/*
+	 * key
+	 */
+	syscall_get_arguments(current, args->regs, 0, 1, &val);
+	res = val_to_ring(args, val, 0, true, 0);
+	if (unlikely(res != PPM_SUCCESS))
+		return res;
+
+	/*
+	 * nsems
+	 */
+	syscall_get_arguments(current, args->regs, 1, 1, &val);
+	res = val_to_ring(args, val, 0, true, 0);
+	if (unlikely(res != PPM_SUCCESS))
+		return res;
+
+	/*
+	 * semflg
+	 */
+	syscall_get_arguments(current, args->regs, 2, 1, &val);
+	res = val_to_ring(args, semget_flags_to_scap(val), 0, true, 0);
+	if (unlikely(res != PPM_SUCCESS))
+		return res;
+
+	return add_sentinel(args);
+}
+
+static int f_sys_semget_x(struct event_filler_arguments *args)
```

Comment:
I think you should be able to use f_sys_single_x here instead of an explicit filler